### PR TITLE
k8s-in-rancher guide for horitzontal-pod-autoscaler

### PR DESCRIPTION
**Pull request by rawmind0**
k8s-in-rancher guide for horitzontal-pod-autoscaler on Rancher v2

### CODE REVIEW

File: content/rancher/v2.x/en/k8s-in-rancher/horitzontal-pod-autoscaler/_index.md [hunk 1]
```diff
@@ -0,0 +1,732 @@
...
+
+Be sure that your k8s cluster services are running at least with these flags: 
+
+- kube-api: `requestheader-client-ca-file` 
+- kubelet: `read-only-port` at 10255
+- kube-controller: Optional, just needed if distinct values than default are required.
+  - `horizontal-pod-autoscaler-downscale-delay: "5m0s"`
+  - `horizontal-pod-autoscaler-upscale-delay: "3m0s"`
+  - `horizontal-pod-autoscaler-sync-period: "30s"`
+
```
Comment (janeczku) [inline content/rancher/v2.x/en/k8s-in-rancher/horitzontal-pod-autoscaler/_index.md]: Add a link to know what these parameters mean: https://rancher.com/docs/rancher/v2.x/en/installation/ha-server-install/

File: content/rancher/v2.x/en/k8s-in-rancher/horitzontal-pod-autoscaler/_index.md [hunk 2]
```diff
@@ -0,0 +1,732 @@
...
+      read-only-port: 10255
+```
+
+Once k8s cluster is configured and deployed properly, is needed to deploy metrics service.
+
+Note: For deploy and test examples, Rancher v2.0.6 and k8s v1.10.1 cluster are being used.
+
+#### Resource metrics 
+
+If hpa wants to use resource metrics, package `metrics-server` is needed at `kube-system` namespace of k8s cluster.
```
Comment (janeczku) [inline content/rancher/v2.x/en/k8s-in-rancher/horitzontal-pod-autoscaler/_index.md]: What about:
> In order to  create horizontal pod autoscaler resources based on resource metrics (e.g. pod CPU/memory usage), you will need to deploy the `metrics-server` in the `kube-system` namespace of k8s cluster, which will enable HPA to consume the `metrics.k8s.io` API.

File: content/rancher/v2.x/en/k8s-in-rancher/horitzontal-pod-autoscaler/_index.md (hunk 1 above)
Comment (rawmind0) [reply content/rancher/v2.x/en/k8s-in-rancher/horitzontal-pod-autoscaler/_index.md]: Added k8s controller manager parameters explanation at `How HPA works` and added link to k8s documentation.

File: content/rancher/v2.x/en/k8s-in-rancher/horitzontal-pod-autoscaler/_index.md (hunk 2 above)
Comment (rawmind0) [reply content/rancher/v2.x/en/k8s-in-rancher/horitzontal-pod-autoscaler/_index.md]: ok

File: content/rancher/v2.x/en/k8s-in-rancher/horitzontal-pod-autoscaler/_index.md [hunk 3]
```diff
@@ -0,0 +1,732 @@
...
+
+  - If you are accessing to k8s cluster throught rancher, server url at kubectl config like `https://<RANCHER_URL>/k8s/clusters/<CLUSTER_ID>` You need to add prefix `/k8s/clusters/<CLUSTER_ID>` to api path
+  ```
+  # kubectl get --raw /k8s/clusters/<CLUSTER_ID>/apis/metrics.k8s.io/v1beta1
+  {"kind":"APIResourceList","apiVersion":"v1","groupVersion":"metrics.k8s.io/v1beta1","resources":[{"name":"nodes","singularName":"","namespaced":false,"kind":"NodeMetrics","verbs":["get","list"]},{"name":"pods","singularName":"","namespaced":true,"kind":"PodMetrics","verbs":["get","list"]}]}
+  ```
+
+#### Custom metrics (prometheus)
+
+Custom metrics could be provided with many third applications as source. We are gonna use [prometheus](https://prometheus.io/) for the example. We are assuming that prometheus is deployed at k8s cluster, getting proper metrics from pods, nodes, namespaces,.... We'll use prometehus url, http://prometheus.mycompany.io exposed at port 80
```
Comment (janeczku) [inline content/rancher/v2.x/en/k8s-in-rancher/horitzontal-pod-autoscaler/_index.md]: Replace:
> Custom metrics could be provided with many third applications as source. 

With:
> Besides acting on resource metrics, HPA can be configured to autoscale based on custom metrics provided by a third party. The most important use case is to be able to autoscale based on application-level metrics (e.g. HTTP requests per second). HPA uses the `custom.metrics.k8s.io` API to consume these metrics. This API is enabled by deploying a custom metrics adapter corresponding to the metrics collection solution.

**Comment by janeczku** [comment]
@rawmind0 LGTM with comments.

File: content/rancher/v2.x/en/k8s-in-rancher/horitzontal-pod-autoscaler/_index.md (hunk 3 above)
Comment (rawmind0) [reply content/rancher/v2.x/en/k8s-in-rancher/horitzontal-pod-autoscaler/_index.md]: ok

**Comment by rawmind0** [comment]
Thanks @janeczku ... Updated PR following your comments...

**Comment by deniseschannon** [comment]
Closing in favor of #626

**Pull request closed without merging**